stable/datadog/templates/daemonset.yaml

```diff
@@ -50,14 +54,24 @@ spec:
         {{- if .Values.datadog.apmEnabled }}
           {{- include "container-trace-agent" . | nindent 8 }}
         {{- end }}
-        {{- if .Values.datadog.processAgentEnabled }}
+        {{- if or (kindIs "invalid" .Values.datadog.processAgentEnabled) .Values.datadog.processAgentEnabled }}
           {{- include "container-process-agent" . | nindent 8 }}
         {{- end }}
-      initContainers:
-        {{ include "containers-init" . | nindent 8 }}
       {{- else }}
         {{- include "container-agents" . | nindent 8 }}
       {{- end}}
+      {{- if .Values.systemProbe.enabled }}
+        {{- include "system-probe-agent" . | nindent 8 }}
+      {{- end }}
+      {{- if or .Values.systemProbe.enabled .Values.daemonset.useDedicatedContainers }}
+      initContainers:
+      {{- if .Values.systemProbe.enabled }}
```

Comment:
1. From my point of view, `systemProbe` always runs as a separate container whether other agents are in one or multiple containers, so I didn't check `useDedicatedContainers`.
2. I don't think so, `system-probe-init` just copies the seccomp profile over, it has nothing to do with any other agents.

Reply:
I feel like system-probe is a great extension to dedicated containers and for the sake of simplicity, readability, usability it would be better to keep that here and not back-porting the feature alongside the all-in-one approach.